presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -836,6 +836,22 @@ public void testInvalidInsert()
                 "INSERT INTO t6 (a, b, c, d) VALUES (1, 1, 1)");
     }
 
+    @Test
+    public void testInvalidInsertNested()
+    {
+        assertFails(MISMATCHED_SET_COLUMN_TYPES,
+                "line 1:29: Mismatch at column 2: 'b.x.z' is of type double but expression is of type varchar.3.",
+                "INSERT INTO t10 VALUES (10, ROW(20, ROW(30, 'abc')))");
+
+        assertFails(MISMATCHED_SET_COLUMN_TYPES,
+                "line 1:29: Mismatch at column 2: 'b.x' has 3 expression.s. but expected 2 fields.s.",
```

Comment:
Modify this similar to the comment above.

Reply:
made changes